### PR TITLE
lig-3121: Update API client docstring - part 3

### PR DESCRIPTION
**Pull request by huan-lightly-0**
Updates docstrings for methods of the API client. This PR covers
* api_workflow_selection.py
* api_workflow_tags.py
* api_workflow_upload_dataset.py
* api_workflow_upload_embeddings.py
* api_workflow_upload_metadata.py

### CODE REVIEW

File: lightly/api/api_workflow_selection.py [hunk 1]
```diff
@@ -33,7 +33,17 @@ def _parse_active_learning_scores(scores: Union[np.ndarray, List]):
 
 
 class _SelectionMixin:
-    def upload_scores(self, al_scores: Dict[str, np.ndarray], query_tag_id: str):
+    def upload_scores(
+        self, al_scores: Dict[str, np.ndarray], query_tag_id: str
```
Comment (guarin) [inline lightly/api/api_workflow_selection.py:37]:
```suggestion
        self, al_scores: Dict[str, NDArray[np.float_]], query_tag_id: str
```

You can import `NDArray` like this:
```
from numpy.typing import NDArray
```

File: lightly/api/api_workflow_selection.py [hunk 2]
```diff
@@ -33,7 +33,17 @@ def _parse_active_learning_scores(scores: Union[np.ndarray, List]):
...
+    ) -> None:
+        """Uploads active learning scores for a tag.
+
+        Args:
+            al_scores: Active learning scores. Must be a mapping between score names
```
Comment (guarin) [inline lightly/api/api_workflow_selection.py:42]: Please put the description on a new line if it spans multiple lines.

File: lightly/api/api_workflow_selection.py [hunk 3]
```diff
@@ -59,26 +69,31 @@ def sampling(self, *args, **kwargs):
...
 
         Returns:
             The newly created tag of the selection.
 
         Raises:
-            ApiException
-            ValueError
-            RuntimeError
+            ApiException:
+                When errors occur during calls to Lightly API.
```
Comment (guarin) [inline lightly/api/api_workflow_selection.py:92]: Should we now include this in all methods that make api calls?

File: lightly/api/api_workflow_upload_dataset.py [hunk 4]
```diff
@@ -61,25 +61,25 @@ def upload_dataset(
         input: Union[str, "LightlyDataset"],
         max_workers: int = 8,
         mode: str = "thumbnails",
-        custom_metadata: Union[Dict, None] = None,
-    ):
-        """Uploads a dataset to to the Lightly cloud solution.
+        custom_metadata: Optional[Dict] = None,
+    ) -> None:
+        """Uploads a dataset to the Lightly Platform.
```
Comment (guarin) [inline lightly/api/api_workflow_upload_dataset.py:66]: Do you know if this endpoint still works? I don't think we support it anymore.

File: lightly/api/api_workflow_upload_metadata.py [hunk 5]
```diff
@@ -37,7 +38,7 @@ def _assert_key_exists_in_custom_metadata(key: str, dictionary: Dict):
 class _UploadCustomMetadataMixin:
     """Mixin of helpers to allow upload of custom metadata."""
 
-    def verify_custom_metadata_format(self, custom_metadata: Dict):
+    def verify_custom_metadata_format(self, custom_metadata: Dict) -> None:
```
Comment (guarin) [inline lightly/api/api_workflow_upload_metadata.py:41]:
```suggestion
    def verify_custom_metadata_format(self, custom_metadata: Dict[str, Any]) -> None:
```

File: lightly/api/api_workflow_upload_dataset.py [hunk 6]
```diff
@@ -61,25 +61,25 @@ def upload_dataset(
         input: Union[str, "LightlyDataset"],
         max_workers: int = 8,
         mode: str = "thumbnails",
-        custom_metadata: Union[Dict, None] = None,
-    ):
-        """Uploads a dataset to to the Lightly cloud solution.
+        custom_metadata: Optional[Dict] = None,
```
Comment (guarin) [inline lightly/api/api_workflow_upload_dataset.py:64]:
```suggestion
        custom_metadata: Optional[Dict[str, Any]] = None,
```

File: lightly/api/api_workflow_upload_metadata.py (hunk 5 above)
Comment (guarin) [reply lightly/api/api_workflow_upload_metadata.py:41]: Please also change it in the methods below.

**Review by guarin — CHANGES_REQUESTED** [review]
Thanks a lot for the PR!

File: lightly/api/api_workflow_upload_dataset.py (hunk 4 above)
Comment (huan-lightly-0) [reply lightly/api/api_workflow_upload_dataset.py:66]: Probably not. I didn't add an E2E test for this.

**Review by guarin — APPROVED** [review]
LGTM! Let's handle the question about `upload_dataset` in a follow-up.

**Pull request merged**